core/frontend/src/IModelConnection.ts

```diff
@@ -557,6 +549,60 @@ export abstract class IModelConnection extends IModel {
         vp.invalidateController();
     }
   }
+
+  /** @internal */
+  public getMapEcefToDb(bimElevationBias: number): Transform {
+    if (!this.ecefLocation)
+      return Transform.createIdentity();
+
+    const mapEcefToDb = this.ecefLocation.getTransform().inverse();
+    if (!mapEcefToDb) {
+      assert(false);
+      return Transform.createIdentity();
+    }
+    mapEcefToDb.origin.z += bimElevationBias;
+
+    return mapEcefToDb;
+  }
+  private _geodeticToSeaLevel?: number;
+  private _projectCenterAltitude?: number;
+
+  /** Event called immediately after map requst is completed.  This occurs only in the case where background map terrain is displayed
+   * with either geiod or ground offset.   These require a query to BingElevation and therefore synching the view may be required
+   * when the request is completed.
+   * @internal
+   */
+  public readonly onMapElevationLoaded = new BeEvent<(_imodel: IModelConnection) => void>();
+
+  /** The offset between sea level and the geodetic ellipsoid.  This will return undefined only if the request for the offset to Bing Elevation
+   * is in process, and in this case the onMapElevationLoaded event is raised when the request is completed.
+   * @internal
+   */
+  public get geodeticToSeaLevel(): number | undefined {

```

Comment:
Every time this or `projectCenterAltitude` is accessed, it dispatches a new request, until one of the requests finally resolves. That could be many many requests. Consider storing a Promise?

Reply:
Yes - good idea.  Done.

Reply:
Would be more manageable with one variable instead of two.
```
private _geodeticToSeaLevel?: number | Promise<number>;
public get geodeticToSeaLevel(): number | undefined {
  if ("number" === typeof this._geodeticToSeaLevel)
    return this._geodeticToSeaLevel;

  if (!this._geodeticToSeaLevel) {
    this._geodeticToSeaLevel = /* ... */
  }

  return undefined;
}
```

Reply:
Maybe - but that seems more confusing to me.

Reply:
With one variable it is clear that there are exactly 3 states you can be in.
With two variables you introduce the possibility of nonsensical states like having both a Promise and a number.

Reply:
OK - Done.